content/posts/hunting-webpack-performances.md

```diff
@@ -0,0 +1,174 @@
+It is very easy to install. ♥️
+
+[Speed Measure Plugin](https://www.npmjs.com/package/speed-measure-webpack-plugin)
+
+Here are the results I got:
+
+```text
+SMP  ⏱  
+General output time took 4 mins, 5.68 secs
+
+ SMP  ⏱  Plugins
+IgnorePlugin took 57.73 secs
+TerserPlugin took 39.022 secs
+ExtractCssChunksPlugin took 3.13 secs
+OptimizeCssAssetsWebpackPlugin took 1.6 secs
+ManifestPlugin took 1.55 secs
+WebpackPwaManifest took 0.326 secs
+ContextReplacementPlugin took 0.129 secs
+HashedModuleIdsPlugin took 0.127 secs
+GenerateSW took 0.059 secs
+DefinePlugin took 0.047 secs
+EnvironmentPlugin took 0.04 secs
+LoadablePlugin took 0.033 secs
+Object took 0.024 secs
+
+ SMP  ⏱  Loaders
+babel-loader, and 
+rev-replace-loader took 2 mins, 11.99 secs
+  module count = 2222
+modules with no loaders took 1 min, 57.86 secs
+  module count = 2071
+extract-css-chunks-webpack-plugin, and 
+css-loader, and 
+postcss-loader, and 
+sass-loader took 1 min, 43.74 secs
+  module count = 95
+css-loader, and 
+postcss-loader, and 
+sass-loader took 1 min, 43.61 secs
+  module count = 95
+file-loader, and 
+rev-replace-loader took 4.86 secs
+  module count = 43
+file-loader took 2.67 secs
+  module count = 32
+raw-loader took 0.446 secs
+  module count = 1
+@bedrock/package-json-loader took 0.005 secs
+  module count = 1
+script-loader took 0.003 secs
+  module count = 1
+```
+
+As expected, it's not great! 
+But at least I'm starting to get who the culprits are.
+We can see that for 2222 Javascript modules takes up 2mins **but** for only 95 Sass files 1min43 🤣.
+
+<iframe src="https://giphy.com/embed/PjNx7g5jtLyJtvDohb" width="480" height="218" frameBorder="0" class="giphy-embed" allowFullScreen></iframe>
+
+### Damn node-sass
+
+Once the migration from `node-sass` to `sass` (new Sass re-implementation) and the update of `sass-loader`, I was shocked!
+It took me about 10 minutes because there were few breaking changes and I **gained more than 1min30** on the build time.
+
+`sass-loader` made big improvements on performances, you should definitely make your sure you use the last version of it.
+
+**I lost a morning on gaining 17 seconds and I spent 10 minutes to win 1min30.🤣**
+
+### IgnorePlugin, TerserPlugin
+
+- `TerserPlugin` is used to uglify the javascript code in order to reduce its size and readability. It's a relatively long process, but *39 seconds* is too much.
+Just by updating the version of TerserPlugin to use the one integrated in Webpack, I could **reduce by 20 seconds** the build time.
+
+- `IgnorePlugin` is a core plugin that was used a lot in our application to avoid loading certain scripts in order to reduce the weight of the site.
+It was necessary, but today with Webpack we can use much better than that. _Dynamic Import, ContextReplacement_, there are plenty of solutions. As a general rule, one should avoid compiling files and then not using them.
```

Comment:
```suggestion
It was necessary, but today with Webpack we can use much better than that. _Dynamic Import, ContextReplacement_, there are plenty of solutions. As a general rule, we should avoid compiling files and then not using them.
```
Though this is very correct, it contrasts a lot with the overall tone of the article 💡 
It's pretty much as if I was saying in french `Pour que l'on comprenne` instead of `Pour qu'on comprenne`